compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -297,16 +298,32 @@ private void replaceMockedFunctions(BIRPackage birPkg, Map<String, String> mockF
             for (BIRBasicBlock functionBasicBlock : functionBasicBlocks) {
                 BIRTerminator bbTerminator = functionBasicBlock.terminator;
                 if (bbTerminator.kind.equals(InstructionKind.CALL)) {
-                    //We get the callee and the name and generate 'calleepackage#name'
                     BIRTerminator.Call callTerminator = (BIRTerminator.Call) bbTerminator;
-                    String functionKey = callTerminator.calleePkg.toString() + MOCK_ANNOTATION_DELIMITER
-                            + callTerminator.name.toString();
+                    String functionKey = null;
+
+                    // Generate the function key
+                    if (callTerminator.isVirtual && callTerminator.args.size() == 1) {
+                        // Function key for object methods
+                        String objectPkg = callTerminator.args.get(0).variableDcl.type.toString();
+                        functionKey = objectPkg + MOCK_ANNOTATION_DELIMITER + callTerminator.name.toString();
+                    } else {
+                        // Function key for normal functions
+                        functionKey = callTerminator.calleePkg.toString() + MOCK_ANNOTATION_DELIMITER
+                                + callTerminator.name.toString();
+                    }
+
+                    // Get the mock function from the Mock function map and replace where necessary
                     if (mockFunctionMap.get(functionKey) != null) {
-                        // Just "get" the reference. If this doesnt work then it doesnt exist
-                        String mockfunctionName = mockFunctionMap.get(functionKey);
-                        callTerminator.name = new Name(mockfunctionName);
+                        String mockFunctionName = mockFunctionMap.get(functionKey);
+                        callTerminator.name = new Name(mockFunctionName);
                         callTerminator.calleePkg = function.pos.src.pkgID;
+
+                        if (callTerminator.isVirtual) {
```

Comment:
Do we need this check. I think line 305 covers it.

Reply:
Actually this is required since this block of code is shared for both types of methods. If this check is not there, then it will process all functions like it is an object method.